File: Daily challenge/Maximize Happiness of Selected Children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) 
    {
        sort(happiness.rbegin(), happiness.rend());
        long long sum = 0;
        for(int i = 0; k-- > 0; i++)
        {
            sum += happiness[i] - i > 0 ? happiness[i] - i : 0;
        }
        return sum;
    }
};

File: Daily challenge/Maximize Happiness of Selected Children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        sum, i = 0, 0
        while k:
            sum += happiness[i] - i if happiness[i] - i > 0 else 0 
            i += 1
            k -= 1
        return sum